core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java

```diff
@@ -101,17 +103,17 @@ public KafkaStore(SchemaRegistryConfig config,
     this.localStore = localStore;
     this.noopKey = noopKey;
 
+    this.config = config;
+
     int zkSessionTimeoutMs =
         config.getInt(SchemaRegistryConfig.KAFKASTORE_ZK_SESSION_TIMEOUT_MS_CONFIG);
     this.zkUtils = ZkUtils.apply(
         kafkaClusterZkUrl, zkSessionTimeoutMs, zkSessionTimeoutMs,
-        JaasUtils.isZkSecurityEnabled());
+        JaasUtils.isZkSecurityEnabled() && this.config.getBoolean(SchemaRegistryConfig.ZOOKEEPER_SET_ACL_CONFIG));
```

Comment:
It would be nicer if these checks were done in a single place that then passed a parameter to other classes.